### ISSUE TITLE
Rake can't drop database unless Rspec has been run first

### ISSUE THREAD

**Issue by agraves** [sporkrb/spork#188]
I've been seeing this issue consistently for a while now.  Basically, when I start spork, if I run "rake" first, I always get this error message:

```
rake aborted!
FOO::Error: ERROR:  database "foo_test" is being accessed by other users
DETAIL:  There are 1 other session(s) using the database.
: DROP DATABASE IF EXISTS "foo_test"
```

If I run any unit test via rspec directly, it will work, and then rake will work every time until I restart the server.

Running:

```
spork (1.0.0rc2)
spork-rails (3.2.0)
```

We're also using database-cleaner, if that matters.

**Comment by fringd** [comment]
this affects me as well...

**Comment by BrandonMathis** [comment]
+1 here too

**Comment by sarahmei** [comment]
This is because rake does a db:test:prepare, which tries to drop the test database, but spork holds on to a connection to the test database after the prefork block completes. I fixed this by adding this line to the bottom of my prefork block:

p "Done setting up. Closing DB connections in test"
ActiveRecord::Base.remove_connection

I did not have to add any connection code to the each_run block - apparently Rails will re-establish whatever connections it needs.
